libcore/FileUtils.vala

```diff
@@ -891,6 +891,81 @@ namespace PF.FileUtils {
 
         return result;
     }
+
+    public string shorten_utf8_string (string base_string, int reduce_by_num_bytes) {
+        var target_length = base_string.length - reduce_by_num_bytes;
+        if (target_length <= 0) {
+            return "";
+        }
+
+        var sb = new StringBuilder.sized (target_length);
+        sb.insert_len (0, base_string, target_length);
+
+        var valid_string = sb.str.make_valid ();
+        if (valid_string.length <= target_length) {
+            return valid_string;
+        } else {
+            return shorten_utf8_string (base_string, reduce_by_num_bytes + 1);
+        }
+    }
+
+    public string get_link_name (string target_name, int count, int max_length = -1)
+    requires (count >= 0) {
+
+        string result = "";
+        switch (count) {
+            case 0:
+                result = target_name; //First link in directory has same name as source
+                break;
+            case 1:
+                result = _("Link to %s").printf (target_name);
+                break;
+            case 2:
+                result = _("Another link to %s").printf (target_name);
+                break;
+
+            case 11:
+                result = _("11th link to %s").printf (target_name);
+                break;
+            case 12:
+                result = _("12th link to %s").printf (target_name);
+                break;
+
+            case 13:
+                result = _("13th link to %s").printf (target_name);
+                break;
+
+            default:
+                break;
+        }
+
+        if (result == "") {
+            switch (count % 10) {
+                case 1:
+                    ///TRANSLATORS: %'d represents 21 or 31 or 41  etc
+                    result = _("%'dst link to %s").printf (count, target_name);
+                    break;
+                case 2:
+                    ///TRANSLATORS: %'d represents 22 or 32 or 42 etc
+                    result = _("%'dnd link to %s").printf (count, target_name);
+                    break;
+                case 3:
+                    ///TRANSLATORS: %'d represents 23 or 33 or 43 etc
+                    result = _("%'drd link to %s").printf (count, target_name);
+                    break;
+                default:
+                    ///TRANSLATORS: %'d represents a number in range 4 - 9, 14 - 19, 24 - 29 etc
+                    result = _("%'dth link to %s").printf (count, target_name);
```

Comment:
This doesn't seem right at all for localization. I'm actually not sure of the right way to do this though. Maybe @ryonakano knows

Reply:
@jeremypw Maybe we can vastly simplify this and go for "FileName (link)", "FileName (link 2)", etc. I think I saw somewhere you were looking at the copy one too. We could apply the same pattern and do "FileName (copy)", "FileName (copy 2)", etc.

Reply:
Yes, I am happy to do that - I was thinking there was a lot of fairly involved code for not very much gain.

Reply:
Naming now simpler